public/docs/dart/latest/guide/forms.jade

```diff
@@ -0,0 +1,662 @@
+:marked
+  We’ve all used a form to log in, submit a help request, place an order, book a flight,
+  schedule a meeting, and perform countless other data entry tasks.
+  Forms are the mainstay of business applications.
+
+  Any seasoned web developer can slap together an HTML form with all the right tags.
+  It's more challenging to create a cohesive data entry experience that guides the
+  user efficiently and effectively through the workflow behind the form.
+
+  *That* takes design skills that are, to be frank, well out of scope for this guide.
+
+  It also takes framework support for
+  **two-way data binding, change tracking, validation, and error handling**
+  ... which we shall cover in this guide on Angular forms.
+
+  We will build a simple form from scratch, one step at a time. Along the way we'll learn:
+
+  - How to build an Angular form with a component and template
+
+  - The `ng-model` two-way data binding syntax for reading and writing values to input controls
+
+  - The `ng-control` directive to track the change state and validity of form controls
+
+  - The special CSS classes that `ng-control` adds to form controls and how to use them to provide strong visual feedback
+
+  - How to display validation errors to users and enable/disable form controls
+
+  - How to share information across controls with template local variables
+
+.l-main-section
+:marked
+  ## Template-driven forms
+
+  Many of us will build forms by writing templates in the Angular
+  template syntax
+  <!-- link to ./template-syntax.html -->
+  with the form-specific directives and techniques described in this guide.
+
+.l-sub-section
+  :marked
+    That's not the only way to create a form but it's the way we'll cover in this guide.
+:marked
+  We can build almost any form we need with an Angular template—login forms, contact forms, pretty much any business form.
+  We can lay out the controls creatively, bind them to data, specify validation rules and display validation errors,
+  conditionally enable or disable specific controls, trigger built-in visual feedback, and much more.
+
+  It will be pretty easy because Angular handles many of the repetitive, boilerplate tasks we'd
+  otherwise wrestle with ourselves.
+
+  We'll discuss and learn to build the following template-driven form:
+
+figure.image-display
+  img(src="/resources/images/devguide/forms/hf-1.png" alt="Clean Form")
+
+:marked
+  Here at the *Hero Employment Agency* we use this form to maintain personal information about the
+  heroes in our stable. Every hero needs a job. It's our company mission to match the right hero with the right crisis!
+
+  Two of the three fields on this form are required. Required fields have a green bar on the left to make them easy to spot.
+
+  If we delete the hero name, the form displays a validation error in an attention-grabbing style:
+
+figure.image-display
+  img(src="/resources/images/devguide/forms/hf-2.png" alt="Invalid, Name Required")
```

Comment:
These form images have always felt to big to me.  After some experimentation, I think 400px works.

Suggest you do as I am doing and replacing `.png" alt`  with `.png" width="400px" alt`   or some such thing.

Review to make sure you like the result and adjust as necessary